### PR TITLE
Feature/ju 313 terraform for ecs

### PR DESCRIPTION
**Pull request by kullas-ch**
### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: terraform/groups/ecs-service/locals.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  name_prefix                = "${local.stack_name}-${var.environment}"
+  global_prefix              = "global-${var.environment}"
+  service_name               = "overseas-entities-api"
+  container_port             = "8080"
+  eric_port                  = "10000"
+  docker_repo                = "overseas-entities-api"
+  kms_alias                  = "alias/${var.aws_profile}/environment-services-kms"
+  lb_listener_rule_priority  = 64
+  lb_listener_paths          = ["/transactions/*/overseas-entity*", "/private/transactions/*/overseas-entity/*"]
+  healthcheck_path           = "/overseas-entity/healthcheck" #healthcheck path for dissolution api
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/locals.tf:13]: Need to update the commend here

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/vars
Comment (chillier-ch) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/vars:1]: This file needs to be in a "staging" directory within the "staging-eu-west-2" directory

File: ecs-image-build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ARG IMAGE_VERSION="latest"
+FROM 416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-runtime-21:${IMAGE_VERSION} 
+ 
+ 
+ #169942020521.dkr.ecr.eu-west-2.amazonaws.com/local/overseas-entities-api
```
Comment (chillier-ch) [inline ecs-image-build/Dockerfile:5]: Can remove this comment

File: terraform/groups/ecs-service/locals.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  stack_name                 = "filing-create" # this must match the stack name the service deploys into
+  name_prefix                = "${local.stack_name}-${var.environment}"
+  global_prefix              = "global-${var.environment}"
+  service_name               = "overseas-entities-api"
+  container_port             = "8080"
+  eric_port                  = "10000"
+  docker_repo                = "overseas-entities-api"
+  kms_alias                  = "alias/${var.aws_profile}/environment-services-kms"
+  lb_listener_rule_priority  = 64
+  lb_listener_paths          = ["/transactions/*/overseas-entity*", "/private/transactions/*/overseas-entity/*"]
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/locals.tf:12]: We also want to add the healthcheck path to the listener paths

File: terraform/groups/ecs-service/locals.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  healthcheck_path           = "/overseas-entity/healthcheck" #healthcheck path for dissolution api
+  healthcheck_matcher        = "200"
+  vpc_name                   = local.stack_secrets["vpc_name"]
+  s3_config_bucket           = data.vault_generic_secret.shared_s3.data["config_bucket_name"]
+  app_environment_filename   = "overseas-entities-api.env"
+  use_set_environment_files  = var.use_set_environment_files
+  application_subnet_ids     = data.aws_subnets.application.ids
+  application_subnet_pattern = local.stack_secrets["application_subnet_pattern"]
+  # dissolutions_bucket_name   = local.service_secrets["dissolutions_bucket_name"]
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/locals.tf:21]: Can be deleted?

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars [hunk 5]
```diff
@@ -0,0 +1,6 @@
+environment = "cidev"
+aws_profile = "development-eu-west-2"
+
+# service configs
+log_level = "debug"
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:5]: We can remove the references to log_level in all the profiles - ideally we just want to define the log level in the ecs-configs env file instead

File: terraform/groups/ecs-service/variables.tf [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  default     = "SSM_VERSION_"
+}
+
+variable "use_set_environment_files" {
+  type        = bool
+  default     = true
+  description = "Toggle default global and shared  environment files"
+}
+
+variable "log_level" {
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/variables.tf:117]: Can remove the log_level variable

**Review by chillier-ch — COMMENTED** [review]
Missing a version file under terraform/groups/ecs-service

**Comment by chillier-ch** [comment]
If there is a parent "java 21 migration" branch then this PR should be directed into that first, and then all changes merged together into main once both dev and platform code is complete

File: terraform/groups/ecs-service/locals.tf [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    trimprefix(sec.name, "/${local.global_prefix}/") => sec.arn
+  }
+
+  global_secret_list = flatten([for key, value in local.global_secrets_arn_map :
+    { "name" = upper(key), "valueFrom" = value }
+  ])
+
+  ssm_global_version_map = [
+    for sec in data.aws_ssm_parameter.global_secret :
+      { "name"  = "GLOBAL_${var.ssm_version_prefix}${replace(upper(basename(sec.name)), "-", "_")}", "value" = tostring(sec.version) }
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/locals.tf:44]: not sure if we want the sec.version to be a string

File: terraform/groups/ecs-service/locals.tf (hunk 1 above)
Comment (kullas-ch) [reply terraform/groups/ecs-service/locals.tf:13]: done

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/vars
Comment (kullas-ch) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/vars:1]: done

File: ecs-image-build/Dockerfile (hunk 2 above)
Comment (kullas-ch) [reply ecs-image-build/Dockerfile:5]: done

File: terraform/groups/ecs-service/locals.tf (hunk 3 above)
Comment (kullas-ch) [reply terraform/groups/ecs-service/locals.tf:12]: done

File: terraform/groups/ecs-service/locals.tf (hunk 4 above)
Comment (kullas-ch) [reply terraform/groups/ecs-service/locals.tf:21]: done. is this not needed for overseas-entities-api?

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars (hunk 5 above)
Comment (kullas-ch) [reply terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:5]: done

File: terraform/groups/ecs-service/variables.tf (hunk 6 above)
Comment (kullas-ch) [reply terraform/groups/ecs-service/variables.tf:117]: done

File: terraform/groups/ecs-service/locals.tf (hunk 4 above)
Comment (chillier-ch) [reply terraform/groups/ecs-service/locals.tf]: I think this was just a specific case for dissolution-api

**Review by chillier-ch — APPROVED** [review]

**Pull request merged**